File: backend/app/schemas/notification.py
from datetime import datetime
from typing import Optional
from uuid import UUID

from pydantic import BaseModel


class NotificationItem(BaseModel):
    id: UUID
    type: str
    title: str
    body: str
    link_url: Optional[str]
    is_read: bool
    created_at: datetime
    archived_at: Optional[datetime] = None


class NotificationListResponse(BaseModel):
    items: list[NotificationItem]
    unread_count: int
    total: int


class BulkIdsRequest(BaseModel):
    ids: list[UUID]

File: frontend/src/types/notification.ts
export type NotificationType =
  | "backtest_completed"
  | "usage_limit_reached"
  | "performance_alert"
  | "new_follower"
  | "strategy_commented"
  | "system";

export interface Notification {
  id: string;
  type: NotificationType;
  title: string;
  body: string;
  link_url: string | null;
  is_read: boolean;
  created_at: string;
}

export interface NotificationListResponse {
  items: Notification[];
  unread_count: number;
}
